frontend/src/core/resource/components/ResourceMenu.js

```diff
@@ -0,0 +1,172 @@
+/* @flow */
+
+import * as React from 'react';
+import { Localized } from 'fluent-react';
+import onClickOutside from 'react-onclickoutside';
+
+import './ResourceMenu.css';
+
+import { ResourceItem, ResourcePercent } from '..';
+
+import type { NavigationParams } from 'core/navigation';
+import type { ResourcesState } from '..';
+
+
+type Props = {|
+    parameters: NavigationParams,
+    resources: ResourcesState,
+    navigateToPath: (string) => void,
+|};
+
+type State = {|
+    search: string,
+    visible: boolean,
+|};
+
+
+/**
+ * Render a resource menu for the main navigation bar.
+ *
+ * Allows to switch between resources without reloading the Translate app.
+ */
+export class ResourceMenuBase extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            search: '',
+            visible: false,
+        };
+    }
+
+    toggleVisibility = () => {
+        this.setState((state) => {
+            return { visible: !state.visible };
+        });
+    }
+
+    // This method is called by the Higher-Order Component `onClickOutside`
+    // when a user clicks outside the user menu.
+    handleClickOutside = () => {
+        this.setState({
+            visible: false,
+        });
+    }
+
+    navigateToPath = (event: SyntheticMouseEvent<>) => {
+        event.preventDefault();
+
+        // $FLOW_IGNORE
```

Comment:
Can you explain why you ignore Flow here?

Reply:
I don't think it's possible to circumvent the "Property pathname is missing in EventTarget [1]." Flow error. Is it?

Reply:
Yeah, I guess so. But can you add a note to say that? :angel:

Reply:
Oh! Well, I can actually get rid of it. :)